From 56b9ca0c60f823904354da3d7147cb14a018d11c Mon Sep 17 00:00:00 2001
From: "dongwan0605.kim" <dongwan0605.kim@lge.com>
Date: Sun, 6 May 2018 17:40:23 +0900
Subject: [PATCH 06/22] Fix intercepting touch events for guts

When guts are showing in keyguard, user's touch events
could be intercepted by ActivatableNotificationView
especially in notification header area.
And then, user had to touch buttons in guts twice to operate.
So, fix not to intercept touch events when guts are showing.

Test: Lockscreen > Swipe EXPANDABLE notification and
click snooze button > Try to click UNDO button or
expand button in snooze layout.

Change-Id: I7f36ca8f38c974561ea7585b4560fff8c2d634fb
---
 .../android/systemui/statusbar/ExpandableNotificationRow.java  | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/ExpandableNotificationRow.java b/packages/SystemUI/src/com/android/systemui/statusbar/ExpandableNotificationRow.java
index b24838fefee..9e73e137403 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/ExpandableNotificationRow.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/ExpandableNotificationRow.java
@@ -2174,6 +2174,9 @@ public class ExpandableNotificationRow extends ActivatableNotificationView
 
     @Override
     protected boolean disallowSingleClick(MotionEvent event) {
+        if (areGutsExposed()) {
+            return false;
+        }
         float x = event.getX();
         float y = event.getY();
         NotificationHeaderView header = getVisibleNotificationHeader();
-- 
2.17.1

